### DISCUSSION TITLE
Why 3 different config files?

### DISCUSSION THREAD

**Discussion by cfinnberg** [archlinux/archinstall#1528] [General]
Hi!

Is there any reasoning to why the settings for automated install are shared into three different config files instead of using one common file? It makes more "difficult" to run the installer:
```
# Why this:
archinstall --config http://myserver.local/config.json --creds http://myserver.local/creds.json --disk_layouts http://myserver.local/disk_layouts.json --silent

# Instead of this:
archinstall --config http://myserver.local/config.json --silent
```

**Comment by Torxed** [comment]
It's due to security reasons.
Users should be able to post `--config` files without risking leaks of sensitive information.
As this is what users do without hesitation we can't include `--creds` for instance, and `--disk-layout` **can** contain disk encryption passwords. It shouldn't, but it can.

That's the short answer :) The long answer is that you can actually have everything stored in `--config` and `archinstall` will accept it I believe. Historically this has been allowed :) It's just that we don't recommend it for the above mentioned reasons.

**Reply by cfinnberg** [reply]
Ok. Thanks for the answer. I will try that, to put everything in the `--config` file. In my use case there are no security concerns.

**Comment by cfinnberg** [comment]
Unfortunately, seems to me that disk layouts can't be put in the same config file. In the `post_process_arguments` of the `__init__.py` file I see that disk layouts are been saved under the key `disk_layouts` ([line 277](https://github.com/archlinux/archinstall/blob/master/archinstall/__init__.py#L277)), but if I add that key myself to the config file, it get caught earlier when trying to parse the command line parameter of the same name ([line 262](https://github.com/archlinux/archinstall/blob/master/archinstall/__init__.py#L262)).

The command line parameter or the dictionary key should change the name so there was no conflict :(

**Reply by Torxed** [reply]
I agree, and this is how it's traditionally been.
That's a regression and something I've approved when in fact I should probably have caught it.
We'll get that sorted out in the future. But now you know the reason and the direction we're heading. Thanks for bringing it up!
*(We should create an issue to track the progress)*
